blockchain/storagev2/leveldb/leveldb_test.go

```diff
@@ -99,16 +102,80 @@ func dirSize(t *testing.T, path string) int64 {
 	return size
 }
 
+func writeBlock(t *testing.T, s *storagev2.Storage, b *types.FullBlock) {
+	t.Helper()
+
+	batchWriter := s.NewWriter()
+
+	batchWriter.PutBody(b.Block.Number(), b.Block.Hash(), b.Block.Body())
+
+	for _, tx := range b.Block.Transactions {
+		batchWriter.PutTxLookup(tx.Hash(), b.Block.Number())
+	}
+
+	batchWriter.PutHeadHash(b.Block.Header.Hash)
+	batchWriter.PutHeadNumber(b.Block.Number())
+	batchWriter.PutBlockLookup(b.Block.Hash(), b.Block.Number())
+	batchWriter.PutHeader(b.Block.Header)
+	batchWriter.PutReceipts(b.Block.Number(), b.Block.Hash(), b.Receipts)
+	batchWriter.PutCanonicalHash(b.Block.Number(), b.Block.Hash())
+	require.NoError(t, batchWriter.WriteBatch())
+}
+
+func readBlock(t *testing.T, s *storagev2.Storage, blockCount int, wg *sync.WaitGroup, ctx context.Context) {
+	t.Helper()
+
+	defer wg.Done()
+
+	ticker := time.NewTicker(20 * time.Millisecond)
+
+	readCount := 1000
+	for i := 1; i <= readCount; i++ {
+		n := uint64(1 + rand.Intn(blockCount))
+
+		hn, ok := s.ReadHeadNumber()
+		if ok && n <= hn {
+			// If head number is read and chain progresed enough to contain canonical block #n
+			h, ok := s.ReadCanonicalHash(n)
+			require.True(t, ok)
+
+			_, err1 := s.ReadBody(n, h)
+			_, err2 := s.ReadHeader(n, h)
+			_, err3 := s.ReadReceipts(n, h)
+			b, err4 := s.ReadBlockLookup(h)
+
+			require.NoError(t, err1)
+			require.NoError(t, err2)
+			require.NoError(t, err3)
+			require.NoError(t, err4)
+
+			assert.Equal(t, n, b)
```

Comment:
It should be better to assert no errors after the function is called. WDYT?

```suggestion
			_, err1 := s.ReadBody(n, h)
			require.NoError(t, err1)
			
			_, err2 := s.ReadHeader(n, h)
			require.NoError(t, err2)
			
			_, err3 := s.ReadReceipts(n, h)
			require.NoError(t, err3)
			
			b, err4 := s.ReadBlockLookup(h)
			require.NoError(t, err4)

			require.Equal(t, n, b)
```

Reply:
Ok, will change this.